shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -0,0 +1,295 @@
+from cloudshell.shell.core.session.cloudshell_session import CloudShellSessionContext
+from cloudshell.shell.core.session.logging_session import LoggingSessionContext    
+#from data_model import *  # run 'shellfoundry generate' to generate data model classes
+
+from googleapiclient import discovery
+from google.oauth2 import service_account
+from google.cloud import storage
+import os
+import json
+
+from constants import GCP_MODELS, GCP1G_MODEL, CREDENTIALS_JSON_PATH, AUTH_URI, TOKEN_URI, TYPE_ACCOUNT, DYNAMIC_JSON, GOOGLE_APPLICATION_CREDENTIALS
+from data_model import GcpTfBackend
+# from contextlib import redirect_stdout
+
+class GcpTfBackendDriver (ResourceDriverInterface):
+
+    def __init__(self):
+        """
+        ctor must be without arguments, it is created with reflection at run time
+        """
+        pass
+
+    def initialize(self, context):
+        """
+        Initialize the driver session, this function is called everytime a new instance of the driver is created
+        This is a good place to load and cache the driver configuration, initiate sessions etc.
+        :param InitCommandContext context: the context the command runs on
+        """
+        pass
+
+    def cleanup(self):
+        """
+        Destroy the driver session, this function is called everytime a driver instance is destroyed
+        This is a good place to close any open sessions, finish writing to log files
+        """
+        pass
+
+    # <editor-fold desc="Discovery">
+
+    def get_inventory(self, context):
+        """
+        Discovers the resource structure and attributes.
+        :param AutoLoadCommandContext context: the context the command runs on
+        :return Attribute and sub-resource information for the Shell resource you can return an AutoLoadDetails object
+        :rtype: AutoLoadDetails
+        """
+        # See below some example code demonstrating how to return the resource structure and attributes
+        # In real life, this code will be preceded by SNMP/other calls to the resource details and will not be static
+        # run 'shellfoundry generate' in order to create classes that represent your data model
+
+        '''
+        resource = GcpTfBackend.create_from_context(context)
+        resource.vendor = 'specify the shell vendor'
+        resource.model = 'specify the shell model'
+        port1 = ResourcePort('Port 1')
+        port1.ipv4_address = '192.168.10.7'
+        resource.add_sub_resource('1', port1)
+        return resource.create_autoload_details()
+        '''
+        gcp_service = self._can_connect(context)
+        if not gcp_service:
+            raise ValueError("Can't connect to GCP")
+        return AutoLoadDetails([], [])
+        os.remove(DYNAMIC_JSON)
+
+    # </editor-fold>
+
+    # <editor-fold desc="Validate S3 Bucket Exists">
+    def _validate_bucket_exists(self, bucket_name, context):
+        with LoggingSessionContext(context) as logger:
```

Comment:
instead of initiating a logger multiple times the best practice is to init a logger at the root and pass it down